pkg/scheduler/actions/preempt/preempt.go

```diff
@@ -289,41 +289,53 @@ func (pmpt *Action) preempt(
 		return pmpt.topologyAwarePreempt(ssn, stmt, preemptor, filter, predicateNodes)
 	}
 
-	return pmpt.normalPreempt(ssn, stmt, preemptor, filter, predicateNodes)
+	targetNodes := predicateNodes
+	needPredicateCheck := false
+	if len(predicateNodes) == 0 {
```

Comment:
`PredicateNodes` not only checks whether resources are sufficient, but also checks such as Volume, Image, NodeAffinity, PodAffinity and other plugins. If all nodes' resources are not enough and we simply preempt all nodes here, what is the meaning of the existence of `PredicateNodes`? I think the correct approach should be to separate resource verification from Volume, Image, NodeAffinity, PodAffinity and other verifications, rather than coupling them in `PredicateNodes`.

Reply:
Thank you for your reply. I agree with your point and will work on improving this.